src/systemctl/systemctl.c

```diff
@@ -7579,10 +7613,36 @@ static int edit(int argc, char *argv[], void *userdata) {
                 goto end;
 
         STRV_FOREACH_PAIR(original, tmp, paths) {
+                _cleanup_free_ char *contents = NULL;
+                const char *contents_marker = NULL;
+                size_t size;
+
+                /* Trim out the lines after the marker */
+                r = read_full_file(*tmp, &contents, &size);
```

Comment:
Should the check for `null_or_empty_path` still be done before this? Asking because if there are any contents in the tmp file at the time of failure (in the code below) it should `goto end` for the cleanup instead of continuing. What do you think?

Reply:
I see the reasoning for the `null_or_empty_path` check - I've added it before the `read_full_file` to do the "temporary file is empty" message.

I'm happy to keep behaviour consistent with how the `rename` handles failure, so I'll do a `goto end` if `read_full_file` fails.
Should reading or renaming fail I think it'd be sensible to simply continue to give a chance to the other units being edited. 
However, such an adjustment in behaviour is unrelated to the goals of this patch (and it'd be a bit more work) so that'd be a job for another patch.